provider/lxd/credentials.go

```diff
@@ -26,6 +56,87 @@ func (environProviderCredentials) DetectCredentials() (*cloud.CloudCredential, e
 }
 
 // FinalizeCredential is part of the environs.ProviderCredentials interface.
-func (environProviderCredentials) FinalizeCredential(_ environs.FinalizeCredentialContext, args environs.FinalizeCredentialParams) (*cloud.Credential, error) {
-	return &args.Credential, nil
+func (p environProviderCredentials) FinalizeCredential(_ environs.FinalizeCredentialContext, args environs.FinalizeCredentialParams) (*cloud.Credential, error) {
+	if args.Credential.AuthType() != cloud.EmptyAuthType {
```

Comment:
Would be helpful to explain when auth type is expected to be "empty" vs "certificate" and that when empty, a certificate auth type is generated yada yada